util/chplenv/chpl_platform.py

```diff
@@ -67,6 +67,16 @@ def get(flag='host'):
 
     return platform_val
 
+@memoize
+def get_linux_distribution():
```

Comment:
IMO `is_fedora` is a better public interface for `chpl_platform` and this could be a helper to implement it. I'm just not sure this will work OK on all linux variants as it is now.